src/lpython/semantics/python_ast_to_asr.cpp

```diff
@@ -139,6 +139,8 @@ ASR::Module_t* load_module(Allocator &al, SymbolTable *symtab,
     return mod2;
 }
 
+std::map<int, ASR::symbol_t*> ast_overload;
```

Comment:
The last part to do is to pass this via arguments to the constructor of the visitor classes and have it as a local variable in `python_ast_to_asr`. Then it will not clash when modules are imported recursively.

Reply:
> Then it will not clash when modules are imported recursively.

That's fine, but I guess it won't be a big issue since we are storing the memory reference of each function which will indeed be unique.

Reply:
Only if you get lucky. When you load multiple modules, the visitor gets created, then destroyed. I think the nodes are allocated using our linear allocator, so I guess they would still be unique. But then when you compile multiple stuff from the main driver and happen to create a new allocator, then it will possibly collide. So we should not rely on that and have no global state.

Reply:
How about using it as a class variable and clearing it completely when a symbol table visitor is instantiated?

Reply:
That will erase it when you import other modules recursively, wouldn't it?

Reply:
Ahh, yes! One probable approach can be to add an optional arguement in `FunctionDef` to store the index where it will be present in `GenericNode`.